internal/config/config.go

```diff
@@ -542,6 +543,15 @@ func (c *Config) IsReleaseImage() bool {
 	return strings.HasPrefix(c.Image, "v")
 }
 
+// IsDebugImage checks whether image name looks like a debug image.
+func (c *Config) IsDebugImage() bool {
```

Comment:
This func should be renamed to make clear that this isn't a reliable check. Otherwise, the use of it can be dangerous depending on what the caller is doing based on the info.

Reply:
what about it is dangerous? you are talking about a case where a customer uses a custom image with exactly our version format: `^ref/([a-zA-Z0-9-]+)/stream/([a-zA-Z0-9-]+)/([a-zA-Z0-9.-]+)$`? and uses debug after `/stream` without it meaning debug image?
That seems unlikely to me, but what name would you suggest then?

Reply:
I'm talking about the opposite case where a debug image is not named like this. This func returning false doesn't mean it's a secure production image.
One name would be the very precise one: `IsNamedLikeDebugImage`. Callers then know that they can use the result of this func for things like printing warnings, but not for checking whether the image has (no) debug capabilities.

Reply:
oki will apply. but just to be sure: debug images are just intended for us developers not for external right? so creating a  debug image is not officially supported anyways right?

Reply:
We might use debug images in PoCs. We might have already done that in the past, not sure.
Creating a debug image from source yourself isn't supported in terms of "Enterprise support", but is valid for external OSS users or external contributors.

IMO it's reason enough that some people may infer so. For example, if you previously used an SGX SDK, you may be used to the meaning of Debug=0 == production enclave.
Apart from the security aspect, functions should not lie/guess. IsDebugImage would need to return "don't know" if NewVersionFromShortPath failed. IsNamedLikeDebugImage can return false without guessing.